core/src/main/java/org/bitcoinj/wallet/WalletEx.java

```diff
@@ -618,23 +622,32 @@ public List<CompactTallyItem> selectCoinsGroupedByAddresses(boolean skipDenomina
     }
 
 
+    /**
+     * Count the number of unspent outputs that have a certain value
+     */
     public int countInputsWithAmount(Coin inputValue) {
         int count = 0;
-        for (Transaction tx : getTransactionPool(WalletTransaction.Pool.UNSPENT).values()) {
-            for (TransactionOutput output : tx.getOutputs()) {
-                if (output.getValue().equals(inputValue) && output.isMine(this))
+        for (TransactionOutput output : myUnspents) {
+            TransactionConfidence confidence = output.getParentTransaction().getConfidence();
+            // confirmations must be 0 or higher, not conflicted or dead
+            if (confidence != null && (confidence.getConfidenceType() == TransactionConfidence.ConfidenceType.PENDING || confidence.getConfidenceType() == TransactionConfidence.ConfidenceType.BUILDING)) {
+                // inputValue must match, the TX is mine and is not spent
+                if (output.getValue().equals(inputValue) && output.getSpentBy() == null) {
                     count++;
+                }
             }
```

Comment:
This fixes a minor bug where spent outputs were counted. Over time this would result in some denominations from being unused, especially 0.001 and 0.01.

Also refactored to turn change this from `O(n^2)` to `O(n)`.

Though the ` output.getSpentBy() == null` is redundant because `myUnspents` would only have unspent outputs.